File: problem44.cpp
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <string>
#include <sstream>
#include <vector>
#include <iostream>
#include <fstream>
#include <omp.h>
using namespace std;
int main()
{
    int N =3000;
   
    #pragma omp parallel for 
    for(int ni=1;ni<N;ni++)
    {
        for(int nj=ni;nj<N;nj++)
        {
            double sum = 3*ni*ni -ni + 3*nj*nj-ni;
            double diff = 3*nj*nj -nj - 3*ni*ni+ni;
            double quadSumm = (1 + sqrt(1+24*sum)) / 12;
            double quadDiff = (1 + sqrt(1+24*diff)) / 12;
            if(fmod(quadSumm,1) == 0 && fmod(quadDiff,1)==0)
                cout << "got one" << endl;
        }
    
    

    }


}

File: problem44.py
import sys
import math

pentNums=[]
for i in range(1,4000):
    pentNums.append(i*(3*i-1)/2)

print len(pentNums)

minimum = sys.maxint
for i in range(0,len(pentNums)):
    for j in range(i+1,len(pentNums)):
        summ = pentNums[i] + pentNums[j]
        diff = pentNums[j] - pentNums[i]

        #if summ in pentNums and diff in pentNums:
        #    print "got one"
        #    print i
        #    print j

        quadSumm = (1 + math.sqrt(1+24*summ)) / 6
        quadDiff = (1 + math.sqrt(1+24*diff)) / 6
        if quadSumm % 1 == 0 and quadDiff % 1 ==0:
            print "got one!"
            print diff

#N = 4000
#for ni in range(1,N):
#    for nj in range(ni,N):
#        summ = 3*ni*ni -ni + 3*nj*nj-ni
#        diff = 3*nj*nj -nj - 3*ni*ni+ni
#        quadSumm = (1 + math.sqrt(1+24*summ)) / 12
#        quadDiff = (1 + math.sqrt(1+24*diff)) / 12
#        if quadSumm == int(quadSumm) and quadDiff == int(quadDiff):
#             print "got one!"
